### DISCUSSION TITLE
api how to set body input

### DISCUSSION THREAD

**Discussion by Pierrelai99** [Budibase/budibase#5235] [Help]
![image](https://user-images.githubusercontent.com/99549872/161552254-7b735457-e6fd-4258-a791-260c38112e2e.png)

hi budibase, I know how to use for input parameters, but this api don't have parameter, only have request body, so how should I create a random input for this OrdrRef ya ? so that I can create a fieldtext then input the text to call this api, thanks for helping me , appreciate

**Comment by melohagan** [comment]
Hi @Pierrelai99 

- [x] It appears you have the first step correct: add a binding **ref** and include it in your request body. 

The next thing you need to do is update the binding in your data provider. See the docs on this here: https://docs.budibase.com/docs/custom-queries

For the value, you can use the following JavaScript:
```
function uuidv4() {
  return ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
    (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
  );
}

return uuidv4();
```
https://stackoverflow.com/questions/105034/how-to-create-a-guid-uuid

**Please do NOT use Math.random() or Date.now(). There's a bug with those.**
